mllib/src/main/scala/org/apache/spark/ml/util/Instrumentation.scala

```diff
@@ -55,6 +55,11 @@ private[spark] class Instrumentation[E <: Estimator[_]] private (
       s" storageLevel=${dataset.getStorageLevel}")
   }
 
+  override def logWarning(msg: => String): Unit = {
+    logNamedValue(Instrumentation.loggerTags.warning, msg)
+    super.logWarning(msg)
```

Comment:
I doubt it will log repeatedly here.

Reply:
Why not add a `level` param into `Instrumentation : def log(msg: String)`. instead of the `logWarning` method.